File: LC682.cpp
class Solution {
public:
    int calPoints(vector<string>& s) {

        stack <int> st;

        int n = s.size();

        for(int i = 0; i < n; i++){\

            if(s[i] == "+"){
                int a = st.top();
                st.pop();
                int res = a + st.top();
                //cout<<res;
                st.push(a);
                st.push(res);
            }

            else if(s[i] == "D")
                st.push(2*st.top());

            else if(s[i] == "C")
                st.pop();
            else
                st.push(stoi(s[i]));
        }

        int sum = 0;

        while(!st.empty()){
            sum += st.top();
            st.pop();
        }

        return sum;
    }
};

File: LC682.py
class Solution:
    def calPoints(self, s: List[str]) -> int:
        l = []

        for i in s:
            if i == "+":
                l.append(l[-1] + l[-2])
            elif i == "C":
                l.pop()
            elif i == "D":
                l.append(2*l[-1])
            else:
                l.append(int(i))
        
        return sum(l)
